File: common-model/src/main/java/com/reservation/commonmodel/auth/Role.java
package com.reservation.commonmodel.auth;

import java.util.Arrays;

public enum Role {
	HOST("ROLE_HOST"),
	CUSTOMER("ROLE_CUSTOMER"),
	ADMIN("ROLE_ADMIN");

	private final String authority;

	Role(String authority) {
		this.authority = authority;
	}

	public String authority() {
		return authority;
	}

	public static Role fromAuthority(String authority) {
		return Arrays.stream(values())
			.filter(r -> r.authority.equals(authority))
			.findFirst()
			.orElseThrow(() -> new IllegalArgumentException("Unknown role: " + authority));
	}
}

File: core/support/src/main/kotlin/com/reservation/support/enums/Role.kt
package com.reservation.support.enums

import java.util.*

enum class Role(private val authority: String) {
    HOST("ROLE_HOST"),
    CUSTOMER("ROLE_CUSTOMER"),
    ADMIN("ROLE_ADMIN");

    fun authority(): String {
        return authority
    }

    companion object {
        fun fromAuthority(authority: String): Role {
            return Arrays.stream(entries.toTypedArray())
                .filter { r: Role -> r.authority == authority }
                .findFirst()
                .orElseThrow {
                    IllegalArgumentException(
                        "Unknown role: $authority"
                    )
                }
        }
    }
}
